packages/core/src/components/menu/_common.scss

```diff
@@ -38,18 +40,62 @@ $dark-menu-background-color: $dark-gray4 !default;
   }
 
   .pt-dark & {
-    @include dark-menu-item($disabled-selector);
+    @include dark-menu-item($disabled-selector, $hover-selector);
   }
 }
 
-@mixin dark-menu-item($disabled-selector: ".pt-disabled") {
+@mixin dark-menu-item($disabled-selector: ".pt-disabled", $hover-selector: ":hover") {
   color: inherit;
 
+  &:not(#{$disabled-selector})#{$hover-selector} {
+    background: $dark-menu-item-color-hover;
+    color: inherit;
+  }
+
   &#{$disabled-selector} {
     color: $pt-dark-text-color-disabled;
   }
 }
 
+@mixin menu-item-intent($text-color: $pt-intent-text-colors) {
+  @each $intent, $color in $text-color {
+    &.pt-intent-#{$intent} {
+      color: $color;
+
+      &::before,
+      &::after,
+      .pt-menu-item-label {
+        color: $color;
```

Comment:
For that loop, can we do it the callout way? https://github.com/palantir/blueprint/blob/master/packages/core/src/components/callout/_callout.scss#L80

The icon should be of the 3rd shade (`$pt-intent-colors`), while the text is always a shade lighter or darker (`$pt-[dark-]intent-text-colors`)